tests/integ/modin/index/test_index_methods.py

```diff
@@ -37,13 +37,46 @@
 ]
 
 
+@pytest.mark.parametrize(
+    "index, type",
+    [
+        (native_pd.Index([1, 2, 3, 4], dtype=int), float),
+        (native_pd.Index(["a", "b", "c"], dtype=str), "object"),
```

Comment:
can you add some more case here to test:
1) string with '1', '2', '3' to int/float
2) boolean to string
3) object with 1, 2, 3 to int